### PR TITLE
Unit test auto runner

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Will automatically run all unit test under the path

t/unittest
## Impacts

Testing t/smoke.t
## NEWS file entries

New Features
++++++++++++

Enhancements
++++++++++++

Bug Fixes
+++++++++
## Delete branch after merge

NO

### CODE REVIEW

**Comment by jrouzierinverse** [comment]
Please review

**Comment by jrouzierinverse** [comment]
Please review

**Comment by jrouzierinverse** [comment]
What is the decision on this?

**Comment by louismunro** [comment]
Put a hold on it.
We’ll merge it if everything else checks out and we still have time.
Otherwise we’ll put it in v5.

## 

LM

**Comment by jrouzierinverse** [comment]
Please Review

File: t/has_unit_test.t [hunk 1]
```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/perl -w
+
+package pf;
+use strict;
+use warnings;
+#use Class::Factory::Util;
```
Comment (cgx) [inline t/has_unit_test.t]: Remove this line if the module won't ever be used.

File: t/example.t [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+use lib '/usr/local/pf/lib';
+
+BEGIN {
+    #include test libs
+    use lib qw(/usr/local/pf/t);
+    #Module for overriding configuration paths
+    use PfFilePaths;
+}
+
+use Test::More tests => 3;
```
Comment (cgx) [inline t/example.t:25]: It looks like there are 2 tests, not 3.

Comment (jrouzierinverse) [inline t/example.t:25]: Test::NoWarnings always adds a test

File: t/TestUtils.pm [hunk 3]
```diff
@@ -222,6 +222,27 @@ sub get_networkdevices_modules {
...
+
+    my @list;
+
+    # find2perl /usr/local/pf/lib/pf/Switch -name "*.pm"
+    File::Find::find({
+        wanted => sub {
+            if ($File::Find::name =~ m#^/usr/local/pf/t/unittest/(.*\.t)$# ) {
+                push(@list, "unittest/$1");
+            }
+        }}, '/usr/local/pf/t/unittest'
```
Comment (fdurand) [inline t/TestUtils.pm:241]: Where is /usr/local/pf/t/unittest ?

**Pull request merged**